docs/architecture-decision-records/002U1-component-constructors.md

```diff
@@ -0,0 +1,125 @@
+# Component Constructors - Update 1
```

Comment:
Might be easier to read [here](https://github.com/guardian/cdk/blob/aa-app-construct/docs/architecture-decision-records/002U1-component-constructors.md)?

The filename was chosen so that it comes under `002-component-constructors.md` in [file browsers](https://github.com/guardian/cdk/tree/aa-app-construct/docs/architecture-decision-records).